cmd/gadgetctl/main.go

```diff
@@ -66,7 +68,23 @@ func main() {
 		// evaluate flags early for runtimeGlobalFlags; this will make
 		// sure that --remote-address has already been parsed when calling
 		// InitDeployInfo(), so it can target the specified address
-		rootCmd.ParseFlags(os.Args[1:])
+
+		// Do not error out on unknown flags, but still validate currently
+		// known ones.
+		// Other flags will be validated in the `Execute()` call and unknown
+		// ones will be rejected
+		rootCmd.FParseErrWhitelist.UnknownFlags = true
+		// temporary workaround for https://github.com/inspektor-gadget/inspektor-gadget/pull/2174#issuecomment-1780923952
+		args := commonutils.RemoveSplitSortArgs(os.Args[1:])
+		// --help should be handled after we registered all commands
+		args = commonutils.RemoveHelpArg(args)
```

Comment:
Mmmmm you mean that if cobra parses here the `--help` flag it will compose immediately the help message even if `Execute` hasn't been called yet? Otherwise, I don't see why we should remove `--help`. What am I missing?

Reply:
The `--help` is not defined (yet) and we get the following error: https://pkg.go.dev/github.com/spf13/pflag@v1.0.5#ErrHelp